src/plugins/condition/components/TestData.vue

```diff
@@ -75,17 +124,117 @@
+            default: () => []
+        },
+        testData: {
+            type: Object,
+            required: true,
+            default: () => {
+                return {
+                    applied: false,
+                    conditionTestInputs: []
+                }
+            }
+        }
     },
     data() {
         return {
             expanded: true,
-            isApplied: true
+            isApplied: false,
+            testInputs: [],
+            telemetryMetadataOptions: {}
         };
     },
+    watch: {
+        isEditing(editing) {
+            if (!editing) {
+                this.resetApplied();
+            }
+        },
+        telemetry: {
+            handler() {
+                this.initializeMetadata();
+            },
+            deep: true
+        },
+        testData: {
+            handler() {
+                this.initialize();
+            },
+            deep: true
+        }
+    },
+    beforeDestroy() {
+        this.resetApplied();
+    },
+    mounted() {
+        this.initialize();
+        this.initializeMetadata();
+    },
     methods: {
-        applyTestData(ev) {
-            this.$emit('change', ev.target.checked);
+        applyTestData() {
+            this.isApplied = !this.isApplied;
+            this.persist();
+        },
+        initialize() {
+            if (this.testData && this.testData.conditionTestInputs) {
+                this.testInputs = this.testData.conditionTestInputs;
+            }
+            if (!this.testInputs.length) {
+                this.addTestInput();
+            }
+        },
+        initializeMetadata() {
+            this.telemetry.forEach(telemetryObj => {
+                this.openmct.objects.get(telemetryObj.identifier).then((telemetryObject) => {
+                    const id = this.openmct.objects.makeKeyString(telemetryObject.identifier);
+                    let telemetryMetadata = this.openmct.telemetry.getMetadata(telemetryObject);
+                    this.telemetryMetadataOptions[id] = telemetryMetadata.values().slice();
+                });
+            });
+        },
+        addTestInput(testInput) {
+            this.testInputs.push(Object.assign({
+                telemetry: '',
+                metadata: '',
+                input: ''
+            }, testInput));
+        },
+        removeTestInput(index) {
+            this.testInputs.splice(index, 1);
+            this.persist();
+        },
+        getId(identifier) {
+            if (identifier) {
+                return this.openmct.objects.makeKeyString(identifier);
+            }
+            return [];
+        },
+        updateMetadata(testInput) {
+            if (testInput.telemetry) {
+                const id = this.openmct.objects.makeKeyString(testInput.telemetry);
+                if(this.telemetryMetadataOptions[id]) {
+                    return;
+                }
+                this.openmct.objects.get(testInput.telemetry).then((telemetryObject) => {
```

Comment:
I'd love it if we weren't fetching the object again here.

Reply:
Removed.